app/src/main/scala/org/alephium/explorer/Main.scala

```diff
@@ -54,67 +54,16 @@ class BootUp extends StrictLogging {
   implicit val databaseConfig: DatabaseConfig[PostgresProfile] =
     DatabaseConfig.forConfig[PostgresProfile]("db", typesafeConfig)
 
-  def init(): Unit =
-    config.bootMode match {
-      case mode: BootMode.Readable =>
-        initInReadableMode(mode)
-
-      case BootMode.WriteOnly =>
-        initInWriteOnlyMode()
-    }
-
-  @SuppressWarnings(Array("org.wartremover.warts.GlobalExecutionContext"))
-  def initInReadableMode(mode: BootMode.Readable): Unit = {
-    //scalastyle:off null
-    var explorer: ExplorerState = null
-    //scalastyle:on null
-    //
-    implicit val executionContext: ExecutionContext =
-      ExecutionContextUtil.fromExecutor(ExecutionContext.global, sideEffect(explorer.stop()))
-
-    explorer = mode match {
-      case BootMode.ReadOnly  => ExplorerState.ReadOnly()
-      case BootMode.ReadWrite => ExplorerState.ReadWrite()
-    }
-
-    explorer
-      .start()
-      .onComplete {
-        case Success(_) => ()
-        case Failure(e) =>
-          logger.error(s"Fatal error during initialization", e)
-          explorer.stop().failed foreach { error =>
-            logger.error("Failed to stop explorer", error)
-          }
-      }
-
-    explorer
-      .start()
-      .onComplete {
-        case Success(_) => ()
-        case Failure(error) =>
-          logger.error("Fatal error during initialization", error)
-          explorer.stop().failed foreach { error =>
-            logger.error("Failed to stop explorer", error)
-          }
-      }
```

Comment:
I didn't realise that on master `explorer.start()` is called twice.

Reply:
ahah yeah my fault on previous refactor. Hopefully our `Service` can't be started twice.